pkg/webui/account/components/oauth-client-form/index.js

```diff
@@ -0,0 +1,346 @@
+      return Yup.string().strip()
+    }
+
+    return Yup.string()
+  }),
+  rights: Yup.array().min(1, sharedMessages.validateRights),
+})
+
+const OAuthClientForm = props => {
+  const {
+    isAdmin,
+    userId,
+    rights,
+    pseudoRights,
+    error,
+    update,
+    initialValues: values,
+    onSubmit,
+    onDelete,
+  } = props
+  const { formatMessage } = useIntl()
+
+  const approvalStateOptions = approvalStates.map(state => ({
+    value: state,
+    label: capitalize(formatMessage({ id: `enum:${state}` })),
+  }))
+
+  const handleSubmit = useCallback(
+    async (values, { resetForm, setSubmitting }) => {
+      const castedValues = validationSchema.cast(values)
+      await onSubmit(castedValues, resetForm, setSubmitting)
+    },
+    [onSubmit],
+  )
+
+  const handleDelete = useCallback(
+    async shouldPurge => {
+      await onDelete(shouldPurge, values.ids.client_id)
+    },
+    [onDelete, values.ids.client_id],
+  )
+
+  const initialValues = {
+    owner_id: userId,
+    rights: [...pseudoRights],
+    ...values,
+  }
+
+  return (
+    <Form
+      error={error}
+      onSubmit={handleSubmit}
+      initialValues={initialValues}
+      validationSchema={validationSchema}
+    >
+      <Form.Field
+        title={sharedMessages.oauthClientId}
+        name="ids.client_id"
+        placeholder={m.clientIdPlaceholder}
+        component={Input}
+        disabled={update}
+        autoFocus={!update}
+        required
+      />
+      <Form.Field
+        title={sharedMessages.name}
+        name="name"
+        placeholder={m.clientNamePlaceholder}
+        component={Input}
+      />
+      <Form.Field
+        title={sharedMessages.description}
+        type="textarea"
+        name="description"
+        placeholder={m.clientDescPlaceholder}
+        description={m.clientDescDescription}
+        component={Input}
+      />
+      <Form.Field
+        name="redirect_uris"
+        title={m.redirectUrls}
+        valuePlaceholder={m.urlsPlaceholder}
+        addMessage={m.addRedirectUri}
+        component={KeyValueMap}
+        indexAsKey
+        description={m.redirectUrlDescription}
+      />
+      <Form.Field
+        name="logout_redirect_uris"
+        title={m.logoutRedirectUrls}
+        valuePlaceholder={m.urlsPlaceholder}
+        addMessage={m.addLogoutRedirectUri}
+        component={KeyValueMap}
+        description={m.logoutRedirectUrlsDescription}
+        indexAsKey
+      />
+      {isAdmin && (
+        <>
+          <Form.Field
+            title={sharedMessages.state}
+            name="state"
+            component={Select}
+            options={approvalStateOptions}
+          />
+          <Form.Field
+            title={sharedMessages.stateDescription}
+            name="state_description"
+            component={Input}
+            type="textarea"
+            placeholder={m.userDescPlaceholder}
+          />
+          <Form.Field
+            title={m.skipAuthorization}
```

Comment:
```suggestion
            label={m.skipAuthorization}
```

Avoid using `title` prop for checkbox fields and use `label`s instead. This will make the form leaner. Please change this for all other occurrences as well (except checkbox groups).